### PR TITLE
One shot action execution

### PR DESCRIPTION
**Pull request by parlt91**
## What does this PR change?

Add support for one shot action execution during create/edit of a recurring custom state schedule. If multiple systems are part of the entity a list that allows for selection of individual systems will be displayed.

## GUI diff

After:

one-shot-button

one-shot-list

- [x] **DONE**

## Documentation

- No documentation needed: Small QoL change. Does not affect the functionality of the feature

- [x] **DONE**

## Test coverage

- No tests: **small visual change.**

- [x] **DONE**

## Links

Issue(s): https://github.com/SUSE/spacewalk/issues/22484

- [x] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

# Before you merge

Check [How to branch and merge properly](https://github.com/uyuni-project/uyuni/wiki/How-to-branch-and-merge-properly)!

### CODE REVIEW

**Review by Etheryte — APPROVED** [review]
The frontend part looks very solid, what do you mean you don't leave me anything to pick at. Nice. 😄

File: java/code/src/com/suse/manager/webui/controllers/RecurringActionController.java [hunk 1]
```diff
@@ -133,6 +144,35 @@ public static String listAll(Request request, Response response, User user) {
...
+                RecurringActionManager.listEntityMembers(type, id, user, pc, PagedSqlQueryBuilder::parseFilterAsText);
+
+        if ("id".equals(pageHelper.getFunction())) {
+            pc.setStart(1);
+            pc.setPageSize(0);
+
+            return json(response, members.stream()
+                    .map(SimpleMinionJson::getId)
+                    .collect(Collectors.toList()));
+        }
```
[the remark this answers is not in the document]
Comment (parlt91) [reply java/code/src/com/suse/manager/webui/controllers/RecurringActionController.java:171]: This gets triggered when you click on the `Select All`. In this case we want to have all the members Ids without paginating them. That's why we set the page back to one and page size to 0 (0 means all the entries regardless of selected page size).

File: java/code/src/com/suse/manager/webui/utils/gson/RecurringActionScheduleJson.java [hunk 2]
```diff
@@ -109,6 +113,15 @@ public Long getId() {
         return getRecurringActionId();
     }
 
+    /**
+     * Gets the List of member ids
+     *
+     * @return the memer ids
```
[the remark this answers is not in the document]
Comment (parlt91) [reply java/code/src/com/suse/manager/webui/utils/gson/RecurringActionScheduleJson.java:119]: Right I'll change it

File: java/code/src/com/suse/manager/webui/controllers/RecurringActionController.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (parlt91) [reply java/code/src/com/suse/manager/webui/controllers/RecurringActionController.java:171]: Ok I get what you're pointing out. You're right I'll have to change it so the query will be called afterwards. Good catch

[the remark this answers is not in the document]
Comment (parlt91) [reply java/code/src/com/suse/manager/webui/controllers/RecurringActionController.java:171]: Exactly my bad

**Pull request merged**